### PR TITLE
Manage preserved tags as a set, with the help of new class PreservedTags

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
    * [ ] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): [PICARD-XXX](https://tickets.metabrainz.org/browse/PICARD-XXX)

# Solution

# Action

### CODE REVIEW

File: picard/ui/metadatabox.py [hunk 1]
```diff
@@ -143,6 +143,36 @@ def tag_status(self, tag):
...
+
+    def __init__(self):
+        self._tags = self._from_config()
+
+    def _to_config(self):
+        config.setting[self.opt_name] =  ", ".join(sorted(self._tags))
+
+    def _from_config(self):
+        return set(
+            [tag for tag in map(lambda x: x.strip(),
```
Comment (sambhav) [inline picard/ui/metadatabox.py]: Not a really big fan of this indentation :P

Comment (sambhav) [reply picard/ui/metadatabox.py]: You can replace lambda exp. with `str.strip`

Comment (sambhav) [reply picard/ui/metadatabox.py]: filter(bool, map(str.strip, config.setting[self.opt_name].split(','))) is prolly what  you want?

Comment (zas) [reply picard/ui/metadatabox.py]: See 7b94c79affb06b1d01bf28dfb649963751cbad13

**Pull request merged**